File: ui/src/eris/types/common/spells/SerializableStranding.ts
import { SpellSchool } from "./SpellSchoolEnum"
import { Type } from "../Type"
import { Who } from "../Who"
import { ISerializableStrandingSpell } from "./ISerializableStrandingSpell"

export class SerializableStranding implements ISerializableStrandingSpell {
    Who: Who
    conjuredType: Type
    guid: string
    title: string
    spellSchool: SpellSchool
    whosName: string
    wasCast: boolean
}

File: core/spells/serialization/SerializableStranding.cs
using System;

namespace Rzeka;
  [Serializable]
public struct SerializableStranding : ISerializableStrandingSpell
{
    public Type conjuredType { get; set; }
    public Guid guid { get; set; }
    public string title { get; set; }
    public SpellSchool spellSchool { get; set; }
    public string whosName { get; set; }
    public bool hasMana { get; set; }
    public Who Who { get; set; }
}
